### PR TITLE
fix: Improved homebrew formula

### LINKED ISSUE #342: Brew tap not accepted in homebrew-core
See: https://github.com/Homebrew/homebrew-core/pull/16943

### PR DESCRIPTION
**Pull request by dominikschulz**
This PR improves the homebrew formula generation pipeline step to include additional fields that will allow to build "proper" formula from source files.

Fixes #342 

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [x] I have read the **CONTRIBUTING** document.
* [ ] `make ci` passes on my machine.
* [x] My change requires a change to the documentation.
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: a0a7ee191f2d71b54d039ddc952dbaf0c11a1d16
  * Commits must be in the following format: **%{type}: %{description}**

Guidelines are available at https://github.com/goreleaser/goreleaser/blob/master/CONTRIBUTING.md#create-a-commit

---
This message was auto-generated by https://gitcop.com

File: internal/artifact/artifact.go [hunk 1]
```diff
@@ -11,6 +12,27 @@ import (
 //go:generate stringer -type=Type
 type Type int
 
+func (t Type) String() string {
```
Comment (caarlos0) [inline internal/artifact/artifact.go]: this should not be needed, `go generate` should create this using `stringer`.

see line 12.

Comment (dominikschulz) [reply internal/artifact/artifact.go]: Ah, now I see. This didn't work after a go get, of course. I'll remove it.

**Comment by dominikschulz** [comment]
I'll try to address your comments and update the branch.

**Comment by dominikschulz** [comment]
@caarlos0 Should be OK for review now.

File: docs/120-homebrew.md [hunk 2]
```diff
@@ -78,6 +83,12 @@ brew:
     system "#{bin}/program --version"
     ...
 
+  # Specify any additional formula content in the special section.
+  # Default is empty.
+  special: |
```
Comment (caarlos0) [inline docs/120-homebrew.md]: this is not a homebrew thing, right?

the idea is to use this to add arbitrary instructions to the formula?

if yes, I think this is the indication of other missing features...

**Comment by caarlos0** [comment]
check #625

File: docs/120-homebrew.md (hunk 2 above)
Comment (dominikschulz) [reply docs/120-homebrew.md]: Yes, that's correct. We used to use this in the past but actually don't use it anymore. So I think for our use case we don't really need it.

**Comment by dominikschulz** [comment]
Thanks, #625 looks pretty complete at first sight. I'll give it a try in the next days :+1:

**Comment by caarlos0** [comment]
ok, so I'll close this in favor of #625 as well

**Pull request closed without merging**